OpenEdXMobile/src/main/java/org/edx/mobile/viewModel/CourseViewModel.kt

```diff
@@ -105,15 +118,116 @@ class CourseViewModel @Inject constructor(
         }
     }
 
+    fun getCourseData(
+        courseId: String,
+        courseComponentId: String?,
+        showProgress: Boolean = false,
+        swipeRefresh: Boolean = false,
+        coursesRequestType: CoursesRequestType = LIVE
+    ) {
+        viewModelScope.launch {
+            _swipeRefresh.postValue(swipeRefresh)
+            _showProgress.postValue(showProgress)
+
+            val courseComponentResult =
+                getCourseComponentResults(courseId, courseComponentId, coursesRequestType)
+
+            courseComponentResult.onSuccess { courseComponent ->
+                _swipeRefresh.postValue(false)
+                _showProgress.postValue(false)
+                if (courseComponent == null) {
+                    var newRequestType: CoursesRequestType = LIVE
+                    if (coursesRequestType == APP_LEVEL_CACHE) {
+                        // Course data is not available in app session cache
+                        newRequestType = PERSISTABLE_CACHE
+                    } else if (coursesRequestType == PERSISTABLE_CACHE || coursesRequestType == STALE) {
+                        // Course data is neither available in app session cache nor available in persistable cache
+                        newRequestType = LIVE
+                    }
+                    getCourseData(
+                        courseId,
+                        courseComponentId,
+                        showProgress = true,
+                        swipeRefresh = false,
+                        coursesRequestType = newRequestType
+                    )
+                } else {
+                    if (coursesRequestType == PERSISTABLE_CACHE) {
+                        // Course data exist in persistable cache
+                        // Send a server call in background for refreshed data
+                        getCourseData(
+                            courseId,
+                            courseComponentId,
+                            showProgress = true,
```

Comment:
Based on this condition, I think s`showProgress` should be `false`.
```
       If (courseComponent != null) {
            // Course data exist in persistable cache
            loadingIndicator.setVisibility(View.GONE);
            // Send a server call in background for refreshed data
            getCourseComponentFromServer(false, false);
        } else {
            // Course data is neither available in app session cache nor available in persistable cache
            getCourseComponentFromServer(true, true);
        }
```

Reply:
good catch, fixed.